x/bank/src/keeper.rs

```diff
@@ -81,6 +82,113 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, AK: AuthKeeper<SK, M>, M: Module> Ban
     }
 }
 
+impl<SK: StoreKey, PSK: ParamsSubspaceKey, AK: AuthKeeper<SK, M>, M: Module>
+    StakingBankKeeper<SK, M> for Keeper<SK, PSK, AK, M>
```

Comment:
I agree that bank module should not depend on `staking` module.

I need some additional information to resolve the issue. 
> I think these methods should be part of the `BankKeeper`

Do you mean that I have to extend gears `BankKeeper` by this methods?

> The staking module could implement the `StakingBankKeeper` on the `BankKeeper`

Do you mean that I have to save current solution except moving of implementation of trait into `staking` module? If yes, then I'll keep main logic in the bank module because the the `bank::Keeper` has no public interfaces for required features.

Reply:
I recommend adding the main logic directly to the `BankKeeper`. The `StakingBankKeeper` implementation can delegate to the methods on the `BankKeeper`.